concourse/steps/update_component_deps.py

```diff
@@ -239,18 +229,45 @@ def greatest_references(
             yield matching_refs[-1]
 
 
+def deserialise_extra_component_references(
```

Comment:
generally-speaking, I would prefer to split into:
a) a function that filter for cref-labels
b) a function that does the deserialisation

generally-speaking, I do not like the pattern of a function that claims to do X and silently skips over elements for which it cannot do X